### PR TITLE
CIRC-1825 New endpoint for allowed service points

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
## Purpose
Introduce a new service point for fetching allowed pickup service points. Contains basic request validation, but no real business logic yet.
Resolves [CIRC-1825](https://issues.folio.org/browse/CIRC-1825).

#### TODOS and Open Questions
- [x] Check logging

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/AllowedServicePointsResource.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    AllowedServicePointsRequest request = new AllowedServicePointsRequest(
+      queryParams.get("requester"),
+      queryParams.get("instance"),
+      queryParams.get("item"));
+
+    return validateRequest(request);
+  }
+
+  private static Result<AllowedServicePointsRequest> validateRequest(AllowedServicePointsRequest request) {
+    log.debug("validateRequest:: request={}", request);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/resources/AllowedServicePointsResource.java:67]:
```suggestion

    log.debug("validateRequest:: parameters: request={}", request);
```

File: src/main/java/org/folio/circulation/resources/AllowedServicePointsResource.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    MultiMap queryParams = routingContext.queryParams();
+    AllowedServicePointsRequest request = new AllowedServicePointsRequest(
+      queryParams.get("requester"),
+      queryParams.get("instance"),
+      queryParams.get("item"));
+
+    return validateRequest(request);
+  }
+
+  private static Result<AllowedServicePointsRequest> validateRequest(AllowedServicePointsRequest request) {
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/resources/AllowedServicePointsResource.java:66]:
```suggestion
  private static Result<AllowedServicePointsRequest> validateRequest(
    AllowedServicePointsRequest request) {
```

File: src/main/java/org/folio/circulation/services/AllowedServicePointsService.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.folio.circulation.domain.AllowedServicePointsRequest;
+import org.folio.circulation.support.results.Result;
+
+public class AllowedServicePointsService {
+  private static final Logger log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+
+  public CompletableFuture<Result<Map<RequestType, Set<String>>>> getAllowedServicePoints(
+    AllowedServicePointsRequest request) {
+
+    log.debug("getAllowedServicePoints:: {}", request);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/services/AllowedServicePointsService.java:23]:
```suggestion
    log.debug("getAllowedServicePoints:: parameters: {}", request);
```

File: src/main/java/org/folio/circulation/resources/AllowedServicePointsResource.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  private static JsonObject toJson(Map<RequestType, Set<String>> allowedServicePoints) {
+    JsonObject responseJson = new JsonObject();
+
+    allowedServicePoints.entrySet()
+      .stream()
+      .filter(entry -> entry.getValue() != null && !entry.getValue().isEmpty())
+      .forEach(entry -> responseJson.put(entry.getKey().getValue(), entry.getValue()));
+
+    return responseJson;
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/resources/AllowedServicePointsResource.java:109]: log result?

**Review by roman-barannyk — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**